pkg/kubelet/kubelet.go

```diff
@@ -893,8 +894,17 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 	return m, nil
 }
 
-// Make the service environment variables for a pod in the given namespace.
-func (kl *Kubelet) makeEnvironmentVariables(ns string, container *api.Container) ([]string, error) {
+func (kl *Kubelet) getDownwardAPIEnvVarMap(pod *api.Pod) map[string]string {
```

Comment:
I don't really like creating an entire object each time.  I'd rather have a resolver object which has a map in it (all the previously resolved env) and then each successive ENV you walk you are a) adding to the internal map, b) overriding the default loader (which is the downward api).  I would want the downward API to be a switch.